widgets/blueprintSources/src/BlueprintSources.tsx

```diff
@@ -1,15 +1,46 @@
-// @ts-nocheck File not migrated fully to TS
 import SplitterLayout from 'react-splitter-layout';
 import Actions from './actions';
 
-export default function BlueprintSources({ data, toolbox, widget }) {
+interface BlueprintTree {
+    children: any;
+    key: string;
+    title: string;
+    isDir: boolean;
+    timestamp: number;
+}
+
+interface BlueprintSourcesProps {
+    data: {
+        blueprintId: string;
+        blueprintTree: BlueprintTree;
+        importedBlueprintIds: string[];
+        importedBlueprintTrees: {
+            timestamp: number;
+            children: any;
+        };
```

Comment:
I think we should define it using `BlueprintTree`:
```suggestion
        importedBlueprintTrees: BlueprintTree[];
```

Reply:
applied: https://github.com/cloudify-cosmo/cloudify-stage/pull/2046/commits/50f940b7f1625ce6830044350f7ee29d768834cd